crates/primitives/src/receipt.rs

```diff
@@ -44,6 +47,78 @@ impl Receipt {
     }
 }
 
+/// A collection of receipts organized as a two-dimensional vector.
+#[derive(Clone, Debug, PartialEq, Eq, Default)]
+pub struct Receipts {
+    /// A two-dimensional vector of optional `Receipt` instances.
+    pub receipt_vec: Vec<Vec<Option<Receipt>>>,
+}
+
+impl Receipts {
+    /// Create a new `Receipts` instance with an empty vector.
+    pub fn new() -> Self {
+        Self { receipt_vec: vec![] }
+    }
+
+    /// Create a new `Receipts` instance from an existing vector.
+    pub fn from_vec(vec: Vec<Vec<Option<Receipt>>>) -> Self {
+        Self { receipt_vec: vec }
+    }
+
+    /// Returns the length of the `Receipts` vector.
+    pub fn len(&self) -> usize {
+        self.receipt_vec.len()
+    }
+
+    /// Returns `true` if the `Receipts` vector is empty.
+    pub fn is_empty(&self) -> bool {
+        self.receipt_vec.is_empty()
+    }
+
+    /// Push a new vector of receipts into the `Receipts` collection.
+    pub fn push(&mut self, receipts: Vec<Option<Receipt>>) {
+        self.receipt_vec.push(receipts);
+    }
+
+    /// Retrieves a vector of references to receipts for the specified index.
+    pub fn root_slow(&self, index: usize) -> Option<Vec<&Receipt>> {
+        Some(self.receipt_vec[index].iter().map(Option::as_ref).collect())?
+    }
+
+    /// Retrieves gas spent by transactions as a vector of tuples (transaction index, gas used).
+    pub fn gas_spent_by_tx(&self) -> Result<Vec<(u64, u64)>, &'static str> {
+        self.last()
+            .map(|block_r| {
+                block_r
+                    .iter()
+                    .enumerate()
+                    .map(|(id, tx_r)| {
+                        if let Some(receipt) = tx_r.as_ref() {
+                            Ok((id as u64, receipt.cumulative_gas_used))
+                        } else {
+                            Err("Receipts have not been pruned")
```

Comment:
error should be `Receipts have been pruned`, since now we're giving out the error and not an expectation

Reply:
@joshieDo Yes sorry, done